Move ephemeral block to top-level sidebar (#488)
hashicorp/web-unified-docs: content/terraform/v1.10.x/docs/language/ephemeral/index.mdx

---
page_title: Ephemeral block reference for the Terraform configuration language
description: Learn to define ephemeral blocks in Terraform configurations to keep temporary and sensitive information out of Terraform state and plan files.
---

# Ephemeral block reference

Ephemeral resources are Terraform resources that are essentially temporary. Ephemeral resources have a unique lifecycle, and Terraform does not store them in its state. Each `ephemeral` block describes one or more ephemeral resources, such as a temporary password or connection to another system.

-> **Note**: Ephemeral resources are available in Terraform v1.10 and later.

## Lifecycle

> The lifecycle of an `ephemeral` resource is different from other resources and data sources. When Terraform provisions ephemeral resources, it performs the following steps:

Author: ```suggestion
The lifecycle of an `ephemeral` resource is different from other resources and data sources. Terraform performs the following lifecycle steps for each ephemeral resource in a configuration:
```

1. If Terraform needs to access the result of an ephemeral resource, it opens

> that ephemeral resource. For example, if Terraform opens an ephemeral resource for a Vault secret, the Vault provider obtains a lease and returns a secret.

Reviewer: ```suggestion
that ephemeral resource. For example, if Terraform opens an ephemeral resource for a Vault secret, the Vault provider obtains a lease and returns the secret.
```

1. If Terraform needs access to the ephemeral resource for longer than the
remote system's enforced expiration time, Terraform asks the provider
to periodically renew it. For example, if Terraform renews a Vault secret `ephemeral` resource, the Vault provider calls Vault's lease renewal API endpoint to extend the expiration time.

1. Once Terraform no longer needs an ephemeral resource, Terraform closes
it. This happens after the providers that depend on an ephemeral resource
complete all of their work for the current Terraform run phase. For example, closing a Vault secret ephemeral resource means the Vault provider explicitly ends the lease, allowing Vault to immediately revoke the associated credentials.

Terraform follows these lifecycle steps for each instance of an ephemeral

> resource in a given configuration.

Reviewer: Maybe this should go before the list?
